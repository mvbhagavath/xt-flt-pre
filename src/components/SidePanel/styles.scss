@import '../../sass/base';

.side-panel {
  background: $white;
  box-shadow: $box-shadow;
  margin: 10px 0 0;
  padding: 10px;
  
  &__filter-label{
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-family: $Medium;
  }

  &__filter-label-icon{
    display: none;
    border-radius: 50%;
    background: transparent;
    min-width: 26px;
    &:focus{
      outline: none;
    }
  }
}

@media (max-width: 768px) {
  .side-panel{
    &__filter-label-icon{
      display: block;
    }

    &__filter-container{
      height: 0;
      overflow: hidden;
      transition: height 2s ease-in-out;
      &.active{
        height: 240px;
      }
    }
  }
}
