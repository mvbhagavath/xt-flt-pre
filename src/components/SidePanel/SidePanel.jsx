import React, { useState } from "react";
import Filters from "../Filters";
import "./styles.scss";

const filters = [
  {
    label: "Species",
    filterItems: ["Human", "Animal", "Alien"],
    id: "species",
  },
  {
    label: "Gender",
    filterItems: ["Male", "Female", "unknown"],
    id: "gender",
  },
  {
    label: "Origin",
    filterItems: ["Earth", "Mars", "unknown"],
    id: "origin",
  },
];

const SidePanel = () => {
  const [isFilterOpen, setFilterState] = useState(false);
  return <div className="col-lg-2 side-panel">
    <div className="side-panel__filter-label">
      <span>Filters</span>
      <button
        className="side-panel__filter-label-icon"
        onClick={() => setFilterState(!isFilterOpen)}
      >
        {isFilterOpen ? '-' : '+'}
      </button>
    </div>
    <div className={`side-panel__filter-container ${isFilterOpen ? 'active' : '' }`}>
      {filters.map((filter) => (
        <Filters key={filter.id} {...filter} />
      ))}
    </div>
  </div>
};

export default SidePanel;
