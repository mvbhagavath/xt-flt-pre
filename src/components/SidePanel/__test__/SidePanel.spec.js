import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "@testing-library/react";
import SidePanel from "..";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  fetchData: {},
});

describe("SidePanel", () => {
  it("should render component without crashing", () => {
    const wrapper = render(
      <Provider store={store}>
        <SidePanel />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
