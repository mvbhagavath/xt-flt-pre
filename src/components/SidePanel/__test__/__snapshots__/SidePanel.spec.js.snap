// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel should render component without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-lg-2 side-panel"
      >
        <div
          class="side-panel__filter-label"
        >
          <span>
            Filters
          </span>
          <button
            class="side-panel__filter-label-icon"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="side-panel__filter-container 
          "
        >
          <div
            class="filter-group"
          >
            <div
              class="filter-group__label"
            >
              Species
            </div>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Human
            </button>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Animal
            </button>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Alien
            </button>
          </div>
          <div
            class="filter-group"
          >
            <div
              class="filter-group__label"
            >
              Gender
            </div>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Male
            </button>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Female
            </button>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              unknown
            </button>
          </div>
          <div
            class="filter-group"
          >
            <div
              class="filter-group__label"
            >
              Origin
            </div>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Earth
            </button>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              Mars
            </button>
            <button
              class="filter-group__filter inactive"
              type="button"
            >
              unknown
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-lg-2 side-panel"
    >
      <div
        class="side-panel__filter-label"
      >
        <span>
          Filters
        </span>
        <button
          class="side-panel__filter-label-icon"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="side-panel__filter-container 
          "
      >
        <div
          class="filter-group"
        >
          <div
            class="filter-group__label"
          >
            Species
          </div>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Human
          </button>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Animal
          </button>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Alien
          </button>
        </div>
        <div
          class="filter-group"
        >
          <div
            class="filter-group__label"
          >
            Gender
          </div>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Male
          </button>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Female
          </button>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            unknown
          </button>
        </div>
        <div
          class="filter-group"
        >
          <div
            class="filter-group__label"
          >
            Origin
          </div>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Earth
          </button>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            Mars
          </button>
          <button
            class="filter-group__filter inactive"
            type="button"
          >
            unknown
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
